src/transaction.rs

```diff
@@ -0,0 +1,220 @@
+use secp256k1::{PublicKey, Signature};
+use crate::{
+    slip::{Slip},
+    time::{create_timestamp}
+};
+
+/// A single record used in the history of transactions being routed around the network
+#[derive(Debug, PartialEq, Clone, Copy)]
+pub struct Hop {
+    pub from: PublicKey,
+    pub to: PublicKey,
+    sig: Signature,
+}
+
+impl Hop {
+    /// Creates a new `Hop`
+    ///
+    /// `to` - `secp256k1::PublicKey` address of where the transaction is headed
+    /// * `from` - `secp256k1::PublicKey` address of where the transaction came from
+    /// * `sign` - `secp256k1::Signature` verifying work done by routers
+    pub fn new(to: PublicKey, from: PublicKey, sig: Signature) -> Hop {
+        return Hop { to, from, sig }
+    }
+}
+/// A record containging data of funds between transfered between public addresses. It
+/// contains additional information as an optinal message field to transfer data around the network
+#[derive(Debug, PartialEq)]
+pub struct Transaction {
+    body: TransactionBody
+}
+
+/// Core data to be serialized/deserialized of `Transaction`
+#[derive(Debug, PartialEq)]
+pub struct TransactionBody {
+    id:       u64,
+    ts:       u64,
+    pub to:   Vec<Slip>,
```

Comment:
My intuition is that slips have to fields, but transactions don't need them.

Reply:
We want to keep `Slip`s as simple as possible, and allow `Transaction`s to do the account balancing between the `to` fields and the `from` fields

Reply:
I had a misunderstanding. Discussed with Stephen. This LGTM.